homeassistant/components/southern_company/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Southern Company integration."""
+from __future__ import annotations
+
+from southern_company_api.parser import SouthernCompanyAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+# For your initial PR, limit it to 1 platform.
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Southern Company from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = SouthernCompanyAPI(
```

Comment:
We should only add it to the `hass.data` once we know we can continue the setup (so test authentication before adding it to the hass data).